torch/nn/modules/module.py

```diff
@@ -398,7 +398,10 @@ def compute_should_use_set_data(tensor, tensor_applied):
 
         for key, buf in self._buffers.items():
             if buf is not None:
-                self._buffers[key] = fn(buf)
+                if isinstance(buf, UninitializedBuffer):
+                    self._buffers[key].data = fn(buf)
```

Comment:
Why do you use .data here?

Reply:
when applying a function an `UninitializedBuffer` it returns a regular `Tensor` object where `materialize` can't be called later.

An alternative is to construct another `UninitializedBuffer` object.

Reply:
Yes I think you want to rewrap it here.
`.data` should not be used anymore.

Reply:
sure! I remember you told me in a different PR, sorry about it